air/src/execution_step/execution_context/context.rs

```diff
@@ -146,12 +148,36 @@ pub(crate) struct ExecCtxIngredients {
 
 #[derive(Debug, Default, Clone)]
 pub struct ExecutionCidState {
-    pub(crate) value_tracker: CidTracker<JValue>,
-    pub(crate) tetraplet_tracker: CidTracker<SecurityTetraplet>,
-    pub(crate) canon_tracker: CidTracker<CanonCidAggregate>,
+    pub value_tracker: CidTracker<JValue>,
+    pub tetraplet_tracker: CidTracker<SecurityTetraplet>,
+    pub canon_tracker: CidTracker<CanonCidAggregate>,
+    pub service_result_agg_tracker: CidTracker<ServiceResultAggregate>,
 }
 
 impl ExecutionCidState {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    pub fn insert_value(
+        &mut self,
+        value: Rc<JValue>,
+        tetraplet: RcSecurityTetraplet,
+        argument_hash: Rc<str>,
+    ) -> Result<Rc<CID<ServiceResultAggregate>>, CidCalculationError> {
```

Comment:
Hmm, just noticed that, but looks like `CidCalculationError` isn't really about calculation and more about se/de issues? Maybe it's worth renaming it?

Reply:
Wouldn't it mean a leaking abstraction then?

Reply:
I got this intention, let's leave it as is.